File: test_archives/cpp_tests/batch5/test_function0.cpp
#include "../catch.hpp"
#include "../functions/function0.cpp"

TEST_CASE( "function0 multiply_result", "[multiply_result]" ) {
    SECTION("Correct results") {
        REQUIRE(function0(1, 2.5, 'test', {2, 3}).multiply_result() == 15);
        REQUIRE(function0(2, 3.5, 'test', {1, 2}).multiply_result() == 14);
        REQUIRE(function0(3, 4.5, 'test', {3, 4}).multiply_result() == 162);
    }
}

File: test_archives/py_tests/batch5/test_function0.py
from functions.function0 import function0
import pytest

@pytest.mark.parametrize('int_input, float_input, list_input, expected', [
    (1, 2.5, [2, 3], 15),
    (2, 3.5, [1, 2], 14),
    (3, 4.5, [3, 4], 162)
])
def test_multiply_result(int_input, float_input, list_input, expected):
    f0 = function0(int_input, float_input, 'test', list_input)
    result = f0.multiply_result()
    assert result == expected